src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1853,6 +1853,11 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
             // Mark completion if we successfully executed all actions and only if we are analyzing a span containing the entire syntax node.
             if (success && analysisStateOpt != null && !declarationAnalysisData.IsPartialAnalysis)
             {
+                // Ensure that we do not mark declaration complete/clear state if cancellation was requested.
+                // Other thread(s) might still be executing analysis, and clearing state could lead to corrupt execution
+                // or unknown exceptions.
```

Comment:
Is there any reasonable way to add a test for this behavior?

Reply:
> Is there any reasonable way to add a test for this behavior?

None that I can think of.

Reply:
Quickly looking through the code this seems like the first time that cancellation is activated on this code path. Perhaps it's done further down and I'm just missing that. Is cancellation being handled properly in all the callers?